servicetalk-http-utils/src/main/java/io/servicetalk/http/utils/HostHeaderHttpRequesterFilter.java

```diff
@@ -42,16 +43,30 @@
  */
 public final class HostHeaderHttpRequesterFilter implements StreamingHttpClientFilterFactory,
                                                             StreamingHttpConnectionFilterFactory {
+
     private final CharSequence fallbackHost;
+    private final boolean useAuthorityOnAbsoluteUri;
 
     /**
      * Create a new instance.
      *
      * @param fallbackHost The address to use as a fallback if a {@link HttpHeaderNames#HOST} header is not present.
      */
     public HostHeaderHttpRequesterFilter(CharSequence fallbackHost) {
+        this(fallbackHost, false);
+    }
+
+    /**
+     * Create a new instance.
+     *
+     * @param fallbackHost The address to use as a fallback if a {@link HttpHeaderNames#HOST} header is not present.
+     * @param useAuthorityOnAbsoluteUri whether to use the request authority (minus user-info) if the request target
+     *                                  is in absolute form.
+     */
+    public HostHeaderHttpRequesterFilter(final CharSequence fallbackHost, final boolean useAuthorityOnAbsoluteUri) {
```

Comment:
It feels like we should always behave like `useAuthorityOnAbsoluteUri == true` bcz this is what spec says. Do we need a public API flag?

Reply:
If we didn't have it exposed as public API I would have agreed without hesitation, but I worry direct users of `HostHeaderHttpRequesterFilter` may already have expectations wrt behavior.

Reply:
Maybe... but it seems more like a bug in `HostHeaderHttpRequesterFilter` that it does not consider absolute-form URL as dictated by spec

Reply:
Done.